tests/contrib/kafka/test_kafka.py

```diff
@@ -287,12 +288,43 @@ def retry_until_not_none(factory):
     return None
 
 
+@pytest.mark.parametrize("payload_and_length", [("test", 4), ("你".encode("utf-8"), 3), (b"test2", 5)])
+@pytest.mark.parametrize("key_and_length", [("test-key", 8), ("你".encode("utf-8"), 3), (b"t2", 2)])
+def test_data_streams_payload_size(
+    dsm_processor, deserializing_consumer, serializing_producer, kafka_topic, payload_and_length, key_and_length
+):
+    payload, payload_length = payload_and_length
+    key, key_length = key_and_length
+    expected_payload_size = float(payload_length + key_length)
+    expected_payload_size += 8  # to account for headers we add here
+    expected_payload_size += len(PROPAGATION_KEY)  # Add in header key length
+    expected_payload_size += 20  # to account for path header we add
```

Comment:
I think the decision to iteratively build up the expected payload size makes this code nicely self-documenting. One could go even further and turn the comments into constant names, though that's a matter of taste and not necessary.